core/src/main/java/co/aikar/commands/CommandReplacements.java

```diff
@@ -61,7 +61,8 @@ public String addReplacement(String key, String val) {
     @Nullable
     private String addReplacement0(String key, String val) {
         key = ACFPatterns.PERCENTAGE.matcher(key.toLowerCase(Locale.ENGLISH)).replaceAll("");
-        Pattern pattern = Pattern.compile("%" + Pattern.quote(key) + "\\b", Pattern.CASE_INSENSITIVE);
+        Pattern pattern = Pattern.compile("%\\{" + Pattern.quote(key) + "}|%" + Pattern.compile(key) + "\\b",
```

Comment:
the 2nd should be Pattern.quote, not Pattern.compile, surprised this didn't error.

and can we move this to ACFPatterns to use the cache method

Reply:
> and can we move this to ACFPatterns to use the cache method

```java
//addReplacement
String pattern = "%\\{" + Pattern.quote(key) + "}|%" + Pattern.quote(key) + "\\b";

//replace
for (Map.Entry<String, String> entry : replacements.values()) { //changed replacements Map key to String, key is the pattern not compiled
    text = ACFPatterns.getPattern(entry.getKey()).matcher(text).replaceAll(entry.getValue());
}
```

You mean something like this?

Reply:
see comment on the other line

Reply:
Hm, already moved then.
The REPLACEMENT_PATTERN to check for unregistered replacements.